From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Mon, 23 May 2016 22:06:09 +0200
Subject: [PATCH] ipmi adapter: Port config support.

Sometimes the IPMI lanplus protocol listens on a non-standard
remote port, e.g. when target nodes are interfaced through a
fake IPMI BMC application that listens on multiple ports on the
same IP address.

Therefore, allow setting IPMI port in DEA/DHA and pass it along
to `ipmitool` when set.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 deploy/dha_adapters/ipmi_adapter.py | 7 +++++--
 deploy/reap.py                      | 3 +++
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/deploy/dha_adapters/ipmi_adapter.py b/deploy/dha_adapters/ipmi_adapter.py
index 283bd57..e806dba 100644
--- a/deploy/dha_adapters/ipmi_adapter.py
+++ b/deploy/dha_adapters/ipmi_adapter.py
@@ -30,12 +30,15 @@ class IpmiAdapter(HardwareAdapter):
         ip = self.get_node_property(node_id, 'ipmiIp')
         username = self.get_node_property(node_id, 'ipmiUser')
         password = self.get_node_property(node_id, 'ipmiPass')
-        return ip, username, password
+        ipmiport = self.get_node_property(node_id, 'ipmiPort')
+        return ip, username, password, ipmiport
 
     def ipmi_cmd(self, node_id):
-        ip, username, password = self.get_access_info(node_id)
+        ip, username, password, ipmiport = self.get_access_info(node_id)
         cmd = 'ipmitool -I lanplus -A password'
         cmd += ' -H %s -U %s -P %s' % (ip, username, password)
+        if ipmiport:
+            cmd += '-p %d' % ipmiport
         return cmd
 
     def get_node_pxe_mac(self, node_id):
diff --git a/deploy/reap.py b/deploy/reap.py
index 6feaf17..1c8d8a6 100755
--- a/deploy/reap.py
+++ b/deploy/reap.py
@@ -59,6 +59,8 @@ adapter:
 #       ipmiIp
 #       ipmiUser
 #       ipmiPass
+#     and you *MAY* provide (optional):
+#       ipmiPort
 #   - libvirt adapter you need to provide:
 #       libvirtName: <whatever>
 #       libvirtTemplate: [libvirt/vms/controller.xml | libvirt/vms/compute.xml]
@@ -190,6 +192,7 @@ class Reap(object):
                  'ipmiIp': None,
                  'ipmiUser': None,
                  'ipmiPass': None,
+                 'ipmiPort': None,
                  'libvirtName': None,
                  'libvirtTemplate': None})
 
